### PR TITLE
List available topics

### PR DESCRIPTION
**Pull request by pladreyt**
## Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

- The user cannot list the available topics in the system.

Issue Link: [Issue](https://trello.com/c/wYqedmtb/33-1-as-a-user-i-should-be-able-to-see-all-the-topics-in-the-system)

## What is the new behavior?

- Now the user can list the available topics in the system.

## Other information

The user must be authenticated in order to list the topics. The topics have a name and an image url.

## Preview

N/A

### CODE REVIEW

File: __tests__/services/topics.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      topic = await factory(Topic)().make();
+    });
+
+    it('returns the topics list', async () => {
+      jest.spyOn(topicRepository, 'find')
+        .mockResolvedValue([topic]);
+      const response = await topicsService.listTopics();
+      expect(response).toBeInstanceOf(Array);
+      expect(response).not.toEqual([]);
+    });
```
Comment (ernes128) [inline __tests__/services/topics.service.spec.ts]: I think you could add a test case here of your service returning an empty topic list, what do you think?

File: src/controllers/topic.controller.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+
+@JsonController('/topics')
+@Service()
+export class TopicController {
+  constructor(private readonly topicsService: TopicsService) { }
+
+  @Authorized()
+  @Get()
+  async index(): Promise<Topic[]> {
```
Comment (ernes128) [inline src/controllers/topic.controller.ts]: I think you could find a more descriptive name for this endpoint, don't you? 

Imagine you are a foreigner and start working on this project and I told you: Pablo, this is the topic controller, what do you think the _index_ function does?

If you struggle to answer this question, then, renaming this could be a good option.

File: src/services/topics.service.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { Service } from 'typedi';
+import { getRepository } from 'typeorm';
+import { Topic } from '@entities/topic.entity';
+
+@Service()
+export class TopicsService {
+  private readonly topicRepository = getRepository<Topic>(Topic);
+
+  listTopics(): Promise<Topic[]> {
```
Comment (ernes128) [inline src/services/topics.service.ts]: You could declare this function async in order to send it to the micro-task queue by force and not delegate this to the owner in using a promise-handling approach, what do you think?

**Review by ernes128 — COMMENTED** [review]
Wow! You improved a lot! I left you some changes but they are not big deal. Good job!

**Review by ernes128 — APPROVED** [review]
Good job! Keep on that way!
